### ISSUE TITLE
get an error, RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Samurais** [joe1chen/nodebb-plugin-search-elasticsearch#5]
Installing this plugin in nodebb v1.0.2, after restarting nodebb, it throws such error.

```
11/4 10:24 [86089] - verbose: [meta/js] nodebb.min.js minification complete
11/4 10:24 [86089] - verbose: [hotswap] Router with id `plugins` replaced successfully
11/4 10:24 [86089] - verbose: [plugins] All plugins reloaded and rerouted
11/4 10:24 [86089] - error: RangeError: Maximum call stack size exceeded
    at Function.EventEmitter.listenerCount (events.js:397:38)
    at Log.listenerCount (/Users/hain/glory/cnodebb/node_modules/nodebb-plugin-search-elasticsearch/node_modules/el
asticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount (/Users/hain/glory/cnodebb/node_modules/nodebb-plugin-search-elasticsearch/node_modules/el
asticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount (/Users/hain/glory/cnodebb/node_modules/nodebb-plugin-search-elasticsearch/node_modules/el
asticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount (/Users/hain/glory/cnodebb/node_modules/nodebb-plugin-search-elasticsearch/node_modules/el
asticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
```

So, after configuring elasticsearch, the search does not works.

```
11/4 10:18 [85521] - error: [plugin/elasticsearch] Could not delete and re-create index. Error: not-conne[280/1914]
express deprecated res.send(status): Use res.sendStatus(status) instead node_modules/nodebb-plugin-search-elasticse
arch/library.js:331:8
11/4 10:19 [85521] - warn: [middleware.admin.isAdmin] deprecation warning, no need to use this from plugins!
11/4 10:19 [85521] - error: [plugin/elasticsearch] Could not delete and re-create index. Error: not-connected
11/4 10:22 [85521] - warn: [middleware.admin.isAdmin] deprecation warning, no need to use this from plugins!
11/4 10:22 [85521] - error: [plugin/elasticsearch] Could not delete and re-create index. Error: not-connected
11/4 10:22 [85521] - warn: [middleware.admin.isAdmin] deprecation warning, no need to use this from plugins!
11/4 10:22 [85521] - error: [plugin/elasticsearch] Could not delete and re-create index. Error: not-connected
11/4 10:23 [85521] - warn: [middleware.admin.isAdmin] deprecation warning, no need to use this from plugins!
11/4 10:23 [85521] - warn: [middleware.admin.isAdmin] deprecation warning, no need to use this from plugins!
11/4 10:23 [85521] - error: [plugin/elasticsearch] Could not delete and re-create index. Error: not-connected
^C11/4 10:24 [85521] - info: [app] Shutdown (SIGTERM/SIGINT) Initialised.
11/4 10:24 [85521] - info: [app] Database connection closed.
11/4 10:24 [85521] - info: [app] Web server closed to connections.
```

**Comment by Samurais** [comment]
I also get a warning 

```
11/4 10:49 [87352] - warn: [plugins/load] The following plugins may not be compatible with your version of NodeBB.
This may cause unintended behaviour or crashing. In the event of an unresponsive NodeBB caused by this plugin, run
`./nodebb reset -p PLUGINNAME` to disable it.
  * nodebb-plugin-search-elasticsearch
```

And the elasticsearch module is too old.

**Comment by q8888620002** [comment]
I update elastic search module within the nodebb-plugin-search-elasticsearch but still 

`3/2 00:26:44 [2958] - info: Routes added
3/2 00:26:44 [2958] - info: NodeBB Ready
3/2 00:26:44 [2958] - info: Enabling 'trust proxy'
3/2 00:26:44 [2958] - info: NodeBB is now listening on: 0.0.0.0:4567
3/2 00:27:18 [2958] - error: /api/search
 TypeError: Cannot read property 'replace' of undefined
    at escapeSpecialChars (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/nodebb-plugin-search-elasticsearch/library.js:19:11)
    at Object.Elasticsearch.search [as method] (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/nodebb-plugin-search-elasticsearch/library.js:229:18)
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/src/plugins/hooks.js:114:12
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:375:13
    at iterate (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:262:13)
    at Object.async.forEachOfSeries.async.eachOfSeries (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:281:9)
    at Object.async.inject.async.foldl.async.reduce (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:374:15)
    at fireFilterHook (/Users/eeeeeeric/Downloads/NodeBB-master-3/src/plugins/hooks.js:106:9)
    at Object.Plugins.fireHook (/Users/eeeeeeric/Downloads/NodeBB-master-3/src/plugins/hooks.js:87:5)
    at Object.search.searchQuery (/Users/eeeeeeric/Downloads/NodeBB-master-3/src/search.js:445:10)
    at async.parallel.pids (/Users/eeeeeeric/Downloads/NodeBB-master-3/src/search.js:61:13)
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:718:13
    at async.forEachOf.async.eachOf (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:233:13)
    at _parallel (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:717:9)
    at Object.async.parallel (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:731:9)
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/src/search.js:58:9
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:726:13
    at /Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:52:16
    at async.forEachOf.async.eachOf (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:236:30)
    at _parallel (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:717:9)
    at Object.async.parallel (/Users/eeeeeeric/Downloads/NodeBB-master-3/node_modules/async/lib/async.js:731:9)
    at searchInContent (/Users/eeeeeeric/Downloads/NodeBB-master-3/src/search.js:46:8)
Thu, 02 Feb 2017 16:35:30 GMT express deprecated res.send(status): Use res.sendStatus(status) instead at node_modules/nodebb-plugin-search-elasticsearch/library.js:331:8`

**Comment by joe1chen** [comment]
Fixed with #7
